### ISSUE TITLE
"Maximum call stack size exceeded" when running grunt-serve

### ISSUE THREAD

**Issue by vedran314** [yeoman/generator-webapp_DEPRECATED#396]
Hi I'm using yeoman with generator-webapp. Here is error that appears after I ran:

```
grunt-serve 
```

Here is error log: 
First 2 lines of code appears in my log for about 300-400 times, so I'm pasting only last line. 

```
(node) warning: Recursive process.nextTick detected. This will break in the next version of node. Please       use setImmediate for recursive deferral.

util.js:35
var str = String(f).replace(formatRegExp, function(x) {
                  ^
RangeError: Maximum call stack size exceeded
```

Info about my nodejs and npm instalation:

```
$ node -v
v0.10.29

$ which nodejs
/usr/bin/nodejs

 $ which npm
/home/alfa/.node/bin/npm
```

Because I'm Ubuntu 14.04 user at my .bashrc I appended  this two lines of code so I can run npm without using sudo (everything works without any errors, tested with gulp and other packages).

```
export PATH=$HOME/.node/bin:$PATH
export NODE_PATH=/usr/lib/nodejs:/usr/lib/node_modules:/usr/share/javascript:/home/alfa/.node/lib/node_modules
```

nodejs and npm installation: just followed instructions from 2nd post from: 
http://stackoverflow.com/questions/16302436/install-nodejs-on-ubuntu-12-10

Thank you for any help you can provide :)

**Comment by eddiemonge** [comment]
Did you change anything else after generating the app?

Try changing the value to something like 65000 following http://stackoverflow.com/questions/11332422/how-can-i-increase-the-maximum-call-stack-size-in-node-js and see if that makes a difference

**Comment by sekl** [comment]
Same problem here on Ubuntu 14.04, Node 0.10.29. Did not change anything except the livereload port in the grunt file.

**Comment by sekl** [comment]
...and turns out increasing memory for my virtual machine running Ubuntu helped.

**Comment by marcdexet-cnrs** [comment]
Same problem today in a Ubuntu 14.04 hosted in virtualbox. It worked yesterday
$ npm -v
1.3.10
$ nodejs -v
v0.10.25

EDIT: After restarting VM (without fixing the call size), it works again.
A actual heinsenbug ?

**Comment by BeOleg** [comment]
Same issue here, any solution?

**Comment by kevva** [comment]
@BeOleg, most likely a memory issue with your system.

**Comment by fabricioguzzo** [comment]
Same problem today on native Ubuntu 14.04 x64:
npm -v
2.1.6

nodejs -v
v0.10.33

grunt --version
grunt-cli v0.1.13
grunt v0.4.5

when I restart the server the problem appears.

grunt serve // ok
<ctrl + c> stop grunt
grunt serve // error

(node) warning: Recursive process.nextTick detected. This will break in the next version of node. Please use setImmediate for recursive deferral.
util.js:35
  var str = String(f).replace(formatRegExp, function(x) {
                      ^
RangeError: Maximum call stack size exceeded

**Comment by gion** [comment]
@chebit I had the same issue and it was triggered by the `watch` task. If that's the case, you can try the solution posted in this [SO thread](http://stackoverflow.com/questions/16748737/grunt-watch-error-waiting-fatal-error-watch-enospc).

**Comment by YonasBerhe** [comment]
@gion thanks that helped.

**Comment by deepaksisodiya** [comment]
I finally figured out a similar problem I was having with spell. I was using

grunt.registerTask('spell', [ 'spell']);
The trick was that Grunt doesn't seem to like the repetition in names. When I switch to

grunt.registerTask('spellCheck', [ 'spell']);
Everything worked as it should.

may this help you

**Comment by lappi-lynx** [comment]
@gion doesn't work with ubuntu 14.04(
@deepaksisodiya where is was the code 'grunt.registerTask('spell', [ 'spell']);'? in Gruntfile.js?

**Comment by robertoallende** [comment]
I solved this issue in Ubuntu 14.04 running:

```
echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p
```

**Comment by vipjeffreylee** [comment]
I solved this issue in Ubuntu 14.04 too, running:
echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p

**Comment by ejcer** [comment]
Yea, this was solved with the above two commands for ubuntu 14.04. No idea why this is so isolated to that distro
